src/modules/internet/Internet.cpp

```diff
@@ -93,21 +93,47 @@ std::string Internet::exampleEmail(std::optional<std::string> firstName, std::op
                        Helper::arrayElement<std::string>(emailExampleHosts));
 }
 
-std::string Internet::password(int length)
+std::string Internet::password(int length, PasswordOptions options)
 {
-    const std::string passwordCharacters =
-        "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789~`!@#$%^&*()_-+={[}]|:;\"'<,>.?/";
+    std::string characters;
 
-    std::string password;
+    if (options.upperLetters) {
+        characters += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
```

Comment:
those upperLetters etc are already defined in src/modules/string/data/Characters.h, you can use them and define not already defined group there

Reply:
done